db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ValueNodeList.java

```diff
@@ -427,10 +431,15 @@ public boolean containsAllConstantNodes()
 
 		for (int index = 0; index < size; index++)
 		{
-			if (! ((ValueNode) elementAt(index) instanceof ConstantNode))
+			ValueNode literalVal = (ValueNode) elementAt(index);
+			if (! (literalVal instanceof ConstantNode))
 			{
 				return false;
 			}
+			else if (literalVal instanceof ListValueNode) {
+				if (! ((ListValueNode)literalVal).containsAllConstantNodes())
```

Comment:
OK, I am moving the else if block inside the if block.  The way this function is used, it should only check for ConstantNodes, not ParameterNodes, because it is used to see when we can eliminate duplicates in the parser.